src/GUI/EFCorePowerTools/Handlers/ReverseEngineer/ReverseEngineerHandler.cs

```diff
@@ -21,360 +21,461 @@ internal class ReverseEngineerHandler
-                    ModelNamespace = modelingOptionsResult.Payload.ModelNamespace,
-                    ProjectRootNamespace = modelingOptionsResult.Payload.Namespace,
-                    UseDatabaseNames = modelingOptionsResult.Payload.UseDatabaseNames,
-                    UseInflector = modelingOptionsResult.Payload.UsePluralizer,
-                    UseLegacyPluralizer = modelingOptionsResult.Payload.UseEf6Pluralizer,
-                    UseSpatial = modelingOptionsResult.Payload.MapSpatialTypes,
-                    UseNodaTime = modelingOptionsResult.Payload.MapNodaTimeTypes,
-                    UseDbContextSplitting = modelingOptionsResult.Payload.UseDbContextSplitting,
-                    UseHandleBars = modelingOptionsResult.Payload.UseHandlebars,
-                    SelectedHandlebarsLanguage = modelingOptionsResult.Payload.SelectedHandlebarsLanguage,
-                    IncludeConnectionString = modelingOptionsResult.Payload.IncludeConnectionString,
-                    SelectedToBeGenerated = modelingOptionsResult.Payload.SelectedToBeGenerated,
-                    UseBoolPropertiesWithoutDefaultSql = modelingOptionsResult.Payload.UseBoolPropertiesWithoutDefaultSql,
-                    UseNullableReferences = modelingOptionsResult.Payload.UseNullableReferences,
-                    UseNoConstructor = modelingOptionsResult.Payload.UseNoConstructor,
-                    UseNoNavigations = modelingOptionsResult.Payload.UseNoNavigations,
-                    Dacpac = dacpacPath,
-                    DefaultDacpacSchema = dacpacSchema,
-                    Tables = pickTablesResult.Payload.Objects.ToList(),
-                    CustomReplacers = pickTablesResult.Payload.CustomReplacers.ToList(),
-                    FilterSchemas = filterSchemas,
-                    Schemas = schemas?.ToList(),
-                    CodeGenerationMode = pickDataSourceResult.Payload.IncludeViews ? CodeGenerationMode.EFCore5 : CodeGenerationMode.EFCore3,
-                    UiHint = pickDataSourceResult.Payload.UiHint,
-                };
-
-                if (options.DatabaseType == DatabaseType.SQLServer
-                    && string.IsNullOrEmpty(options.Dacpac))
-                {
-                    var rightsAndVersion = reverseEngineerHelper.HasSqlServerViewDefinitionRightsAndVersion(options.ConnectionString);
+        private async System.Threading.Tasks.Task InstallEfCorePackageAsync(Project project, ReverseEngineerOptions options, Tuple<bool, string> containsEfCoreReference)
+        {
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-                    if (rightsAndVersion.Item1 == false)
-                    {
-                        EnvDteHelper.ShowMessage("The SQL Server user does not have 'VIEW DEFINITION' rights, default constraints may not be available.");
-                    }
+            if (containsEfCoreReference.Item1)
```

Comment:
This should at least be inverted, but the old logic must be preserved if possible:

`if (modelingOptionsResult.Payload.InstallNuGetPackage)`

Reply:
please verify again, the method VerifySQLServerRightsAndVersion have the same code

Reply:
ok, i see now

Reply:
I did a refresh, and the installed package was updated - basically Refresh should probably never install the provider package.